integration-tests/mongodb-client/src/main/java/io/quarkus/it/mongodb/pojo/OptionalPropertyCodecProvider.java

```diff
@@ -0,0 +1,61 @@
+package io.quarkus.it.mongodb.pojo;
+
+import java.util.Optional;
+
+import org.bson.BsonReader;
+import org.bson.BsonWriter;
+import org.bson.codecs.Codec;
+import org.bson.codecs.DecoderContext;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.pojo.PropertyCodecProvider;
+import org.bson.codecs.pojo.PropertyCodecRegistry;
+import org.bson.codecs.pojo.TypeWithTypeParameters;
+
+/**
+ * This PropertyCodecProvider register a codec provider for the Optional type.
+ * This allows the POJO Codec to handle it as it's a generic type.
+ * See: http://mongodb.github.io/mongo-java-driver/3.12/bson/pojos/#generics-support
+ */
+public class OptionalPropertyCodecProvider implements PropertyCodecProvider {
```

Comment:
Does this class get used anywhere?

Reply:
Yes, this is the purpose of this PR ;)

It will be discovered by the processor and used when creating the POJO Codec.
Maybe I need to provie a better javadoc comment to explain this.

Reply:
Ah OK, now I see...

This is part of the integration test, not the extension itself. I thought this was part of the extension and was wondering how it got registered...

So I suggest that you enhance the comment of this class to make it evident that it gets discovered  and automatically registered by the extension - I am sure we'll forget in a while :)